WordPress/gutenberg
# Polish create template part modal

## What?

This PR makes some changes to the create template modal to make it more consistent with other modals.

## Why? / How?

- Simplify the title: The current title is wordy and the multi-line text is a bit hard to read.
- Change the height of the text field and button to 40px
- Remove the default value from the text field and apply a placeholder instead: Templates, patterns and page create modals do not have default values.
- Add "Cancel" button: I think all modals need a cancel button unless there is a special reason not to.
- Apply `small` size to the modal
- Remove unused classname

## Testing Instructions

- Insert the Template Part or Header/Footer block.
  Note: If you have enabled TT4, the Header template part already exists, so even if you insert a Header block, no placeholder will be displayed. It's a good idea to test it with Emptytheme.
- Click the "Start blank" button.
- Creating template parts should still work as before.

## Screenshots or screencast 

| Before | After |
|--------|--------|
| ![template_part_before](https://github.com/user-attachments/assets/8b210a05-dca4-4184-9331-cc7337279100) | ![image](https://github.com/user-attachments/assets/2849b290-0d20-4b03-8e16-84c4cdf93af0) |
| ![header_before](https://github.com/user-attachments/assets/1fda3c24-11fb-4b70-9d35-e9fa7ccb8e19) | ![image](https://github.com/user-attachments/assets/7bf2615a-23ef-46bd-a4ed-358a82f87f8e) |
